### PR TITLE
Support mounting files with --files using secrets.

### PR DESCRIPTION
**Pull request by mccheah**
Supports SPARK-23153 but only for small local files. Ported from [what we used to do](https://github.com/palantir/spark/blob/9a5536817973dd9db12c9384c66ec6224168588e/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/submitsteps/MountSmallLocalFilesStep.scala).

### CODE REVIEW

**Comment by robert3005** [comment]
Can you rebase so we know if the tests are passing?

**Comment by mccheah** [comment]
Discussed in https://github.com/palantir/spark/pull/324 that this should have the circle 2.0 changes.

**Comment by mccheah** [comment]
Will target master after #324 merges.

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala [hunk 1]
```diff
@@ -141,6 +141,38 @@ private[spark] object Config extends Logging {
       .checkValue(interval => interval > 0, s"Logging interval must be a positive time value.")
       .createWithDefaultString("1s")
 
+  val JARS_DOWNLOAD_LOCATION =
```
Comment (robert3005) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala]: you're not actually using this afaik

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/steps/DependencyResolutionStep.scala [hunk 2]
```diff
@@ -33,10 +34,12 @@ private[spark] class DependencyResolutionStep(
     sparkFiles: Seq[String]) extends DriverConfigurationStep {
 
   override def configureDriver(driverSpec: KubernetesDriverSpec): KubernetesDriverSpec = {
-    val resolvedSparkJars = KubernetesUtils.resolveFileUrisAndPath(sparkJars)
-    val resolvedSparkFiles = KubernetesUtils.resolveFileUrisAndPath(sparkFiles)
-
     val sparkConf = driverSpec.driverSparkConf.clone()
+    val resolvedSparkJars = KubernetesUtils.resolveFileUrisAndPath(
```
Comment (robert3005) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/steps/DependencyResolutionStep.scala]: Or at least this doesn't make sense with the constructor of this class

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/steps/DependencyResolutionStep.scala [hunk 3]
```diff
@@ -29,33 +25,17 @@ import org.apache.spark.deploy.k8s.submit.KubernetesDriverSpec
...
 
   override def configureDriver(driverSpec: KubernetesDriverSpec): KubernetesDriverSpec = {
-    val resolvedSparkJars = KubernetesUtils.resolveFileUrisAndPath(sparkJars)
-    val resolvedSparkFiles = KubernetesUtils.resolveFileUrisAndPath(sparkFiles)
-
     val sparkConf = driverSpec.driverSparkConf.clone()
-    if (resolvedSparkJars.nonEmpty) {
-      sparkConf.set("spark.jars", resolvedSparkJars.mkString(","))
-    }
+    val resolvedSparkFiles = KubernetesUtils.resolveFileUrisAndPath(
```
Comment (mccheah) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/steps/DependencyResolutionStep.scala]: This code is meant to be pretty short-lived until we get local file support in upstream.

**Comment by mccheah** [comment]
The build passed - good to merge here?

**Pull request merged**